statsmodels/censored/robustros.py

```diff
@@ -0,0 +1,449 @@
+        This is used after the linear regression to revert the transformation
+        performed by `transform_in`. If `None` is passed, the data will remain
+        untransformed.
+
+    Attributes
+    ----------
+    nobs : int
+        Total number of results in the dataset
+    ncen : int
+        Total number of non-detect results in the dataset.
+    cohn : pandas DataFrame
+        A DataFrame of the unique detection limits found in `data` along
+        with the `A`, `B`, `C`, and `PE` quantities computed by the
+        estimation.
+    data : pandas DataFrame
+        An expanded version of the original dataset `data` passed the
+        constructor included in the `modeled` column.
+    debug : pandas DataFrame
+        A full version of the `data` DataFrame that includes preliminary
+        quantities.
+
+    Example
+    -------
+    >>> from statsmodels.censored.robustros import RobustROSEstimator
+    >>> ros = RobustROSFitter(myDataFrame, result_col='conc',
+                              censorship_col='censored')
+    >>> ros.fit()
+    >>> print(ros.data)
+
+    Notes
+    -----
+    It is inappropriate to replace specific left-censored values with
+    the estimated values from this method. The estimated values
+    (self.data['modeled']) should instead be used to refine descriptive
+    statistics of the dataset as a whole.
+
+    '''
+    def __init__(self, data, result_col='res', censorship_col='cen',
+                 transform_in=np.log, transform_out=np.exp):
+
+        # confirm a datatype real quick
+
+        if not isinstance(data, pd.DataFrame):
+            raise ValueError("Input `data` must be a pandas DataFrame")
+
+        if not data.index.is_unique:
+            raise ValueError("Index of input DataFrame `data` must be unique")
+
+        try:
+            np.float64(data[result_col])
+        except ValueError:
+            raise ValueError('Result data is not uniformly numeric')
+
+        if data[result_col].min() <= 0:
+            raise ValueError('All result values of `data` must be positive')
+
+        self.result_col = result_col
+        self.censorship_col = censorship_col
+
+        # pre- and post- regression transformations
+        self.transform_in = (
+            transform_in if transform_in is not None else lambda x: x
+        )
+        self.transform_out = (
+            transform_out if transform_out is not None else lambda x: x
+        )
+
+        # and get the basic info
+        self.nobs = data.shape[0]
+        self.ncen = data[data[self.censorship_col]].shape[0]
+
+        # sort the data
+        self.data = _ros_sort(
+            data,
+            result_col=self.result_col,
+            censorship_col=self.censorship_col
+        )
+
+        # create a dataframe of detection limits and their parameters
+        # used in the ROS estimation
+        self.cohn = self._get_cohn_numbers()
+
+    def _get_cohn_numbers(self):
```

Comment:
@josef-pkt 

Check the literature and the NADA documentation, A, B, C are described as:

```python
# The Cohn Numbers are:
# A_j   = the number of uncensored obs above the jth threshold.
# B_j   = the number of observations (cen & uncen) below the jth threshold.
# C_j   = the number of censored observations at the jth threshold
# P_j   = the probability of exceeding the jth threshold
```

We could probably just document them as such, or rename name.

So we could try: 
`A` :arrow_right: `_uncensored_above`
`B` :arrow_right: `_obs_below`
`C` :arrow_right: `_censored_equal`
`P` :arrow_right: `_exceedance_prob`

Open to suggestions, though.